MusicTaste/MusicTaste/View Controllers/Login/LoginViewController.m

```diff
@@ -1,98 +1,72 @@
-                                                                       delegate:self];
+- (void)viewDidLoad {
     [super viewDidLoad];
+    //Make Username and Password UIText Delegates
+    self.usernameField.delegate = self;
+    self.passwordField.delegate = self;
 }
 
-#pragma mark - Actions
-
-- (void)didTapAuthButton:(ConnectButton *)sender
-{
-    SpotifyAPIManager *api = [SpotifyAPIManager shared];
-    [api setUpSpotifyWithCompletion:^(NSDictionary *data, NSError *error) {
-        if (!error) {
-            [self performSegueWithIdentifier:@"loginSegue" sender:nil];
-        }else{}
+- (IBAction)logIn:(id)sender {
+    
+    NSString *username = self.usernameField.text;
+    NSString *password = self.passwordField.text;
+    if ([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""]){
+        //If the username and password are empty return Invalid Login
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Invalid Login"
+                                                                       message:@"Username and password required."
+                                                                       preferredStyle:(UIAlertControllerStyleAlert)];
+        //If the username and password are not empty say it is Okay
+        UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
+                                                           style:UIAlertActionStyleDefault
+                                                         handler:^(UIAlertAction * _Nonnull action) {
+                                                         }];
+        [alert addAction:okAction];
+        [self presentViewController:alert animated:YES completion:^{
+        }];
+    }
+    
+    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError *  error) {
+        if (error == nil) {
+            //If the username and password are in the data base perform segue to MatchmakingViewController
+            [self performSegueWithIdentifier:@"tabBarSegue" sender:nil];
+        }
+        else{
+            //If the username and password are not in the data base return Invalid Login UIAlert
+            UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Invalid Login"
+                                                                           message:@"Username and/or password incorrect."
+                                                                           preferredStyle:(UIAlertControllerStyleAlert)];
+            UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
+                                                               style:UIAlertActionStyleDefault
+                                                             handler:^(UIAlertAction * _Nonnull action) {
+                                                             }];
+            [alert addAction:okAction];
```

Comment:
Comparing old LoginViewController to new one

Reply:
Yep, it's hard to review those parts of the pull request when that happens. Normally git auto-detects file renames. But if you move/rename a file _and also_ make significant changes to the file in the same commit then it can't properly detect the move/rename. There are two approaches:

1. Move/Rename in one commit, then make the significant changes in a separate commit OR
2. Use the `git mv` (from the terminal, as described [here](https://www.educative.io/answers/how-to-rename-or-move-files-in-git)) to explicitly have git move/rename the file.

Either of those approaches will facilitate a clean pull request that is easy to review. Next time you're in that situation give it a try and let us know if you run into any problems.